### PR TITLE
cephadm: `cephadm ls` broken for SUSE downstream alertmanager container

### PR DESCRIPTION
**Pull request by p-se**
Fixes an error with `cephadm ls` when SUSE's downstream alertmanager is used. The difference here is that the executable is named `prometheus-alertmanager` instead of `alertmanager` as in the Prometheus upstream image.
```
#==[ Command ]======================================#
# /usr/sbin/cephadm ls
Non-zero exit code 127 from /usr/bin/podman exec 81581144900c6b7f5a1da7124bd0debb26cd455349f162729b4ea3e9f7fc4f86 alertmanager --version
/usr/bin/podman:stderr Error: exec failed: container_linux.go:349: starting container process caused "exec: \"alertmanager\": executable file not found in $PATH": OCI runtime command not found error
[
...
    {
        "style": "cephadm:v1",
        "name": "alertmanager.ses-5-admin",
        "fsid": "33ec5433-261d-3dad-b75a-cd21d35a0946",
        "systemd_unit": "ceph-33ec5433-261d-3dad-b75a-cd21d35a0946@alertmanager.ses-5-admin",
        "enabled": true,
        "state": "running",
        "container_id": "81581144900c6b7f5a1da7124bd0debb26cd455349f162729b4ea3e9f7fc4f86",
        "container_image_name": "foo.bar.de:5000/registry.suse.com/caasp/v4.5/prometheus-alertmanager:0.16.2",
        "container_image_id": "4683615b36cb2f4fb8580ae7ebbe0cdf42db25ae79c7ca3f48f5096706c25679",
        "version": null,
        "started": "2021-02-23T17:25:13.927267",
        "created": "2021-01-20T17:02:49.987979",
        "deployed": "2021-01-22T15:49:37.746184",
        "configured": "2021-01-22T15:49:38.822172" 
    },
...
]
```

Fixes: https://tracker.ceph.com/issues/49506

Signed-off-by: Patrick Seidensal <pseidensal@suse.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -4563,9 +4563,19 @@ def list_daemons(ctx, detail=True, legacy_dir=None):
...
+                                                    container_path, 'exec', container_id, cmd,
+                                                    '--version'
+                                                ], verbosity=CallVerbosity.SILENT)
+                                            if code == 0:
+                                                break
+                                        cmd = 'alertmanager'  # reset cmd for version extraction
+                                    else:
+                                        out, err, code = call(ctx, [
+                                            container_path, 'exec', container_id, cmd, '--version'
+                                        ])
```
Comment (mgfritch) [inline src/cephadm/cephadm:4578]: this function is already becoming incredibly hard to read ... 

any chance we can move the version logic up into a staticmethod of the `Monitoring` class?

https://github.com/ceph/ceph/blob/80fb7538756820c5d996734e21935ae6fdff14cb/src/cephadm/cephadm#L362-L373

**Review by mgfritch — COMMENTED** [review]
this function is already becoming incredibly hard to read ... 

any chance we can move the version logic up into a staticmethod of the `Monitoring` class?

https://github.com/ceph/ceph/blob/80fb7538756820c5d996734e21935ae6fdff14cb/src/cephadm/cephadm#L362-L373

**Comment by p-se** [comment]
jenkins test make check

**Review by mgfritch — APPROVED** [review]
lgtm!

**Review by sebastian-philipp — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2021-03-02_10:27:34-rados:cephadm-wip-swagner-testing-2021-03-01-1657-distro-basic-smithi/

**Pull request merged**